zilliqa/src/api/types/eth.rs

```diff
@@ -88,7 +87,7 @@ pub struct Header {
 }
 
 impl Header {
-    pub fn from_header(header: message::BlockHeader, miner: Address) -> Self {
+    pub fn from_header(header: message::BlockHeader, miner: Address, block_gas_limit: u64) -> Self {
```

Comment:
I do think its important to keep this as an `EvmGas` for as far as we can. It would be easy to confuse a ScillaGas parameter here.